### ISSUE TITLE
Windows X86 (32bit) release?

### ISSUE THREAD

**Issue by PavelDurov1488** [stenzek/duckstation#1489]
Cause its universal and may be wider gamepad support 
I dont know, old school guys will love it

Don't think, that 'superfast' emulator is cool, any normal CPU should to emulate 33mhz...

It's fun that you are focused on upscaling+pgxp, now I prefer to use DS as RA core cause shaders

![image](https://user-images.githubusercontent.com/74456272/105477491-6754ee80-5cfe-11eb-8ff6-2fcc0b91a91f.png)

RA+DuckStation - huge shader pack, autoconfig for gamepads, no dinput vibration
RA+pcsxREARMED - like RA+DS, but i like it for 1x resolution cause of dithering
DS - original emulator, for me it was unusable until I delete some dll files for gamepad driver and unset vsync for HD4000 and software render
ePSXe - dinput vibration, not very cool GPU plugin
xebra - fast, 32bit, OpenGL 2.0, no shaders support, difficult

Maybe its hard and ASM code, but like competition to beat epsxe in any case... In future

**Comment by Margen67** [comment]
Duplicate of #1226

**Comment by stenzek** [comment]
Please stop.

1. DuckStation has nothing to do with Retroarch, do not post about that rubbish here.
2. I've explained time and time again that I have no interest in writing a recompiler for 15 year old hardware.
3. Duckstation is an accurate emulator, it has higher requirements than these ancient emulators which you're talking about. But on modern hardware, it flies, and even outperforms these older emulators. But it's written and optimized for modern hardware, NOT old hardware.

In other words, if you want to use old, unsupported operating systems on old hardware, there's plenty of other emulators out there.

**Comment by PavelDurov1488** [comment]
Eh, but like competition maybe 
As i know, android's PCSX-ReARMed for 32bit works faster than 64

Something like 64bit instructions are faster than 32bit, but 32bit sounds enough for 33mhz CPU, better potential dinput support, but it needs works with asm

I dont know how upscaling works, it seems like developer like upscaling+pgpxp, maybe 64bit upscaling+pgxp is better

Dont know, I prefer 'original experience' 320x240 with good shaders is enough....

https://user-images.githubusercontent.com/74456272/105479567-198db580-5d01-11eb-9ac0-afd882482ee1.mp4

**Comment by PavelDurov1488** [comment]
No problem, I wish you good luck and be healthy anyway

I saw DS on russian side, somebody likes it

![image](https://user-images.githubusercontent.com/74456272/105480386-2d85e700-5d02-11eb-83dd-7fb45ed3368e.png)

**Comment by stenzek** [comment]
You couldn't be more incorrect. pcsxrearmed is faster on 32-bit because it uses a completely different dynarec, optimized like crazy for armv7. 64-bit uses the generic lightrec.

Point-in-case: DuckStation is measurably faster on AArch64 than AArch32 because it allows more modern techniques (e.g. fastmem) and generates better code.

I have limited time to work on this project, and every new platform adds additional maintenance burden and time. I'd rather be working on new/interesting features, rather than fixing up a 32-bit recompiler when it breaks for 2 users.

As I said, there's plenty of alternatives out there, one emulator cannot satisfy all use cases. Especially when it's basically a one-man project.

**Comment by PavelDurov1488** [comment]
If Android 32bit needs 1.5GHz cpu, desktop/laptop should support it anyway (Intel Atom has 1.6Ghz)

![image](https://user-images.githubusercontent.com/74456272/105481809-0cbe9100-5d04-11eb-86b7-55f5ab040cfe.png)

Android 32bit should have much wider user range

It's a little ironic like you hate ancient hardware, but work on cpu from 1995

Upscaling without PGXP increases texture jitter, with PGXP its not authenic. 

https://user-images.githubusercontent.com/74456272/105487988-755e3b80-5d0d-11eb-8201-bec69ef271f2.mp4

But technically yeah, DS suggests best upscaling+pgxp+perfomance solution

**Comment by stenzek** [comment]
I believe a user successfully ran duckstation on a 1.1ghz Celeron, so yes, indeed, such hardware can run it fine. But even this Celeron was 64bit.

ARM 32-bit is much less limiting than 32bit x86, and there's still a decent number of used with moderately specced phones who can make use of it (hundreds/thousands). I can't say the same for 32bit x86, I don't support windows 7, and I doubt there's many 32bit windows 10 users.

I don't hate old hardware, I just have limited time to work on this project, therefore I have to limit the scope.

If you want to employ me full time to work on duckstation, I'll gladly write a 32bit recompiler or whatever you like. Or if you write one, I'm happy to review and merge it. But otherwise it's a free project and we're all volunteers, therefore we decide the scope and what we work on.

**Comment by PavelDurov1488** [comment]
The time, right, and effictivity!

Mostly its philosophy question

For example, its a funny video 'Cyberpunk 2077 but it's for PS1'

https://www.youtube.com/watch?v=SNqWPtcXk5w
![image](https://user-images.githubusercontent.com/74456272/105536858-8169ed80-5d4d-11eb-984b-e36e8ce6f599.png)

It has ~240p look, texture jittering, textures without filtration, cars disappers like its no much memory. Looks funny

If process this with all DuckStation enhancement like upscale+pgxp+textures filtes, its not so funny

Modern philosophy is not about authentic.romantic, but its like like 'faster, more, modern'

DS has Win10 64bit only support not because, for example, XP or win7 8.1 can't play it, but because it's modern. And best and modern upscaling methods, texture replacement

From another side, 240p original PS1 picture looks pretty ugly

Colors are wrong, gamma is wrong, ugly dithering

Its not because original graphic is ugly, but cause no '**psycological**' CRT emulation with color fixes, blur, some mask

RA has huge CRT shader pack, but another error is no calibration of them. If I take a photo from my x768 laptop display, its like

![IMG_20210122_165424](https://user-images.githubusercontent.com/74456272/105539174-ebd05d00-5d50-11eb-9007-bf8086e3b63d.jpg)

I don't see so huge mask and white is more white, black is more black. So, even I want x768 display emulator on 8k display, mask should not be so intensive

Hmm....

**Comment by Margen67** [comment]
@PavelDurov1488 What is your CPU?
